File: src/main/java/sharev/member/repository/MemberRepository.java
package sharev.member.repository;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import sharev.account.entity.Account;
import sharev.member.entity.Member;
import sharev.team.entity.Team;

public interface MemberRepository extends JpaRepository<Member, Long> {
    List<Member> findAllByAccount(Account account);

    Optional<Member> findByTeamAndAccount(Team team, Account account);
}

File: src/main/kotlin/sharev/member/adapter/outbound/jpa/repository/MemberRepository.kt
package sharev.member.adapter.outbound.jpa.repository

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import sharev.member.adapter.outbound.jpa.entity.MemberJpaEntity
import sharev.member.domain.model.MemberRole
import sharev.team.adapter.outbound.jpa.entity.TeamJpaEntity

interface MemberRepository : JpaRepository<MemberJpaEntity, Long> {

    fun findByTeamAndAccountId(team: TeamJpaEntity, accountId: Long): MemberJpaEntity?

    fun existsByTeamIdAndAccountId(teamId: Long, accountId: Long): Boolean

    @EntityGraph(attributePaths = ["account"])
    fun findAllByTeam(team: TeamJpaEntity): List<MemberJpaEntity>

    fun countByTeamAndRole(team: TeamJpaEntity, role: MemberRole): Long
}
